src/plugins/dashboard/public/dashboard_api/open_save_modal.tsx

```diff
@@ -0,0 +1,170 @@
+import { getDashboardContentManagementService } from '../services/dashboard_content_management_service';
+import { DashboardState } from './types';
+import { DASHBOARD_CONTENT_ID, SAVED_OBJECT_POST_TIME } from '../dashboard_constants';
+import { extractTitleAndCount } from '../dashboard_container/embeddable/api/lib/extract_title_and_count';
+import { DashboardSaveModal } from '../dashboard_container/embeddable/api/overlays/save_modal';
+
+/**
+ * @description exclusively for user directed dashboard save actions, also
+ * accounts for scenarios of cloning elastic managed dashboard into user managed dashboards
+ */
+export async function openSaveModal({
+  controlGroupReferences,
+  dashboardState,
+  isManaged,
+  lastSavedId,
+  panelReferences,
+  viewMode,
+}: {
+  controlGroupReferences?: Reference[];
+  dashboardState: DashboardState;
+  isManaged: boolean;
+  lastSavedId: string | undefined;
+  panelReferences: Reference[];
+  viewMode: ViewMode;
+}) {
+  if (viewMode === 'edit' && isManaged) {
+    return undefined;
+  }
+  const dashboardContentManagementService = getDashboardContentManagementService();
+  const saveAsTitle = lastSavedId
+    ? await getSaveAsTitle(dashboardState.title)
+    : dashboardState.title;
+  return new Promise<(SaveDashboardReturn & { savedState: DashboardState }) | undefined>(
+    (resolve, reject) => {
+      const onSaveAttempt = async ({
+        newTags,
+        newTitle,
+        newDescription,
+        newCopyOnSave,
+        newTimeRestore,
+        onTitleDuplicate,
+        isTitleDuplicateConfirmed,
+      }: DashboardSaveOptions): Promise<SaveDashboardReturn> => {
+        const saveOptions = {
+          confirmOverwrite: false,
+          isTitleDuplicateConfirmed,
+          onTitleDuplicate,
+          saveAsCopy: lastSavedId ? true : newCopyOnSave,
+        };
+
+        try {
+          if (
+            !(await dashboardContentManagementService.checkForDuplicateDashboardTitle({
+              title: newTitle,
+              onTitleDuplicate,
+              lastSavedTitle: dashboardState.title,
+              copyOnSave: saveOptions.saveAsCopy,
+              isTitleDuplicateConfirmed,
+            }))
+          ) {
+            return {};
+          }
+
+          const dashboardStateToSave: DashboardState = {
+            ...dashboardState,
+            title: newTitle,
+            tags: savedObjectsTaggingService && newTags ? newTags : ([] as string[]),
+            description: newDescription,
+            timeRestore: newTimeRestore,
+            timeRange: newTimeRestore
+              ? dataService.query.timefilter.timefilter.getTime()
+              : undefined,
+            refreshInterval: newTimeRestore
+              ? dataService.query.timefilter.timefilter.getRefreshInterval()
+              : undefined,
+          };
+
+          // TODO If this is a managed dashboard - unlink all by reference embeddables on clone
```

Comment:
Maybe we link to https://github.com/elastic/kibana/issues/190138 for this TODO?

Reply:
resolved with 4ee0ad50f34be8f0450b81a6b7e2f577b04fce98